src/vanilla/utils/splitAtom.ts

```diff
@@ -111,11 +111,13 @@ export function splitAtom<Item, Key>(
             const nextItem = isFunction(update)
               ? update(arr[index] as Item)
               : update
-            set(arrAtom as WritableAtom<Item[], [Item[]], void>, [
-              ...arr.slice(0, index),
-              nextItem,
-              ...arr.slice(index + 1),
-            ])
+            if (!Object.is(nextItem, arr[index])) {
```

Comment:
Just a minor preference:
```suggestion
            if (!Object.is(arr[index], nextItem)) {
```